src/inmanta/agent/handler.py

```diff
@@ -470,19 +490,45 @@ def run() -> None:
 
         return f.result()
 
-    def set_cache(self, cache: AgentCache) -> None:
-        self.cache = cache
-
     def get_client(self) -> protocol.SessionClient:
         """
         Get the client instance that identifies itself with the agent session.
-
         :return: A client that is associated with the session of the agent that executes this handler.
         """
         if self._client is None:
             self._client = protocol.SessionClient("agent", self._agent.sessionid)
         return self._client
 
+
+@stable_api
+class ResourceHandler(HandlerABC):
+    """
+    A baseclass for classes that handle resources. New handler are registered with the
+    :func:`~inmanta.agent.handler.provider` decorator.
+
+    The implementation of a handler should use the ``self._io`` instance to execute io operations. This io objects
+    makes abstraction of local or remote operations. See :class:`~inmanta.agent.io.local.LocalIO` for the available
+    operations.
+
+    :param agent: The agent that is executing this handler.
+    :param io: The io object to use.
+    """
+
+    def __init__(self, agent: "inmanta.agent.agent.AgentInstance", io: Optional["IOBase"] = None) -> None:
+        self._agent = agent
```

Comment:
Shouldn't we call super here instead of setting `self._agent`, `self._ioloop` and `self._client` directly?